photon-core/src/main/java/org/photonvision/common/networking/NetworkManager.java

```diff
@@ -258,14 +256,22 @@ private void setConnectionStatic(NetworkConfig config) {
     }
 
     // Detects changes in the carrier and reinitializes after re-connect
-    private Runnable deviceStatus(String devName) {
-        Path file = Path.of("/sys/class/net/{device}/carrier".replace("{device}", devName));
-        logger.debug("Watching network interface at path: " + file.toString());
+    private void monitorDevice(String devName, int millisInterval) {
+        String taskName = "deviceStatus-" + devName;
+        if (TimedTaskManager.getInstance().taskActive(taskName)) {
+            // task is already running
+            return;
+        }
+        Path path = Paths.get("/sys/class/net/{device}/carrier".replace("{device}", devName));
```

Comment:
Out of scope/not: we could probably ask Linux to watch this file and notify us if it changes but polling is easy here in Java land and doesn't require knowing Linux programming.

Reply:
Yes. I've also looked into the Java approach for monitoring files and that may eventually be something I implement. But not today.